synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -338,17 +337,10 @@ async def action_for_event_by_user(
                 # current user, it'll be added to the dict later.
                 actions_by_user[uid] = []
 
-            for rule, enabled in rules.rules():
-                if not enabled:
-                    continue
-
-                matches = evaluator.check_conditions(rule.conditions, uid, display_name)
-                if matches:
-                    actions = [x for x in rule.actions if x != "dont_notify"]
-                    if actions and "notify" in actions:
-                        # Push rules say we should notify the user of this event
-                        actions_by_user[uid] = actions
-                    break
+            actions = evaluator.run(rules, uid, display_name)
+            if actions and "notify" in actions:
```

Comment:
why not:
```suggestion
            if "notify" in actions:
```
am I missing something?

Reply:
Ah, I think this used to be `Option<Vec<Action>>`, will update.